notus/scanner/scanner.py

```diff
@@ -109,37 +126,99 @@ def _start_scan(
                         advisory=package_advisory.advisory,
                     )
 
+    def scan_host_ssh(self, msg: ScanHostMessage) -> None:
+        """Handles Data given with ScanHostMessage and runs scan. The Message
+        must contain credentials to setup a SSH connection to get the required
+        os release and package list. If no SSH cannot connect to the host, it
+        just get skipped."""
+        ssh = SSHClient()
+        ssh.set_missing_host_key_policy(AutoAddPolicy())
```

Comment:
This is more a questions: could we fetch host public keys and verify them somehow?

Reply:
AutoAddPolicy is considered a security issue by codeql scans. @y0urself therefore implemented a ask the user solution in python-gvm https://github.com/greenbone/python-gvm/pull/486 i don't think this change is applicable here but maybe it is an inspiration.

Reply:
We could do it configurable, set it to `RejectPolicy()` by default and document it.